include/parsed_mapped_functions.h

```diff
@@ -0,0 +1,181 @@
+#ifndef __dealii_sak_parsed_mapped_function_h
+#define __dealii_sak_parsed_mapped_function_h
+
+#include <deal.II/base/exceptions.h>
+#include "parameter_acceptor.h"
+#include "parsed_function.h"
+#include <deal.II/fe/component_mask.h>
+#include <algorithm>
+#include <map>
+
+using namespace dealii;
+
+/**
+ * ParsedMappedFunctions object.
+ * It allows you to set a mapped functions, i.e., parsed functions
+ * acting on specified id (boundary_id, material_id,etc..) and on
+ * specified components.
+ *
+ * Dirichlet Boundary conditions or Neumann boundary conditions
+ *
+ * can be set using this class.
+ */
+template <int spacedim, int n_components>
+class ParsedMappedFunctions : public ParameterAcceptor
+{
+public:
+//  ParsedMappedFunctions  () {};
+  ParsedMappedFunctions  (const std::string &name = "",
+                          const std::string &component_names = "u",
+                          const std::string &default_id_components = "", // 0: 0; 1; 2 , 1: 3
+                          const std::string &default_id_functions = "",  // 0: x; y; z, 1: sin(k)
+                          const std::string &default_constants = "");
+
+  shared_ptr<ParsedFunction<spacedim,n_components> > get_mapped_function (const unsigned int &id) const;
+
+  ComponentMask get_mapped_mask (const unsigned int &id) const;
+
+  std::vector<unsigned int> get_mapped_ids() const;
+
+
+  virtual void declare_parameters (ParameterHandler &prm);
+
+  /// An entry with this id does not exist in this object.
+  DeclException1(ExcIdNotFound, unsigned int,
+                 << "No entry with the id " << arg1 << " exists.");
+private:
+
+  void split_id_components(const std::string &parsed_id_components);
+  void split_id_functions(const std::string &parsed_id_functions,
+                          const std::string &constants);
+
+
+  std::string name;
+  std::vector<std::string> component_names;
+  std::vector<unsigned int> id_defined_components;
+  std::vector<unsigned int> id_defined_functions;
+  std::map<unsigned int, ComponentMask> id_components;
+  std::map<unsigned int, shared_ptr<ParsedFunction<spacedim,n_components> > > id_functions;
```

Comment:
These should not be `ParsedFunctions` objects. They are parsed **at construction**, so this won't work, unless you only want to specify the functions as an argument to the constructor. 

Use, instead, this class (which is what `ParsedFunction` uses itself):

https://www.dealii.org/8.2.1/doxygen/deal.II/classFunctions_1_1ParsedFunction.html

This will require a little trick when you construct the new objects:

~~~
ParameterHandler internal_prm;
dealii::Functions::ParsedFunction<dim,spacedim>::declare_parameters(internal_prm, n_components);
internal_prm.set("Function expression") = ...
... ptr = new dealii::Functions::ParsedFunction<dim,spacedim>(n_components);
ptr->parse_parameters(internal_prm);
~~~

and you should do this for each function. In this way, if you change parameter file, this is reflected on the functions, which can be in number arbitrary.